### PR TITLE
misc: drop support for PHP 8.0

### PR DESCRIPTION
**Pull request by romm**
PHP 8.0 security support has ended on the 26th of November 2023.

See: https://www.php.net/supported-versions.php

### CODE REVIEW

File: src/Mapper/ArgumentsMapperError.php
Comment (TimWolla) [inline src/Mapper/ArgumentsMapperError.php:1]: The removal of the braces around the variable in the strings is inconsistent with the changes in the other files.

File: src/Mapper/Object/Exception/ObjectBuildersCollision.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ public function __construct(ClassDefinition $class, ObjectBuilder ...$builders)
         $constructors = implode('`, `', $constructors);
 
         parent::__construct(
-            "A collision was detected between the following constructors of the class `{$class->type()->toString()}`: `$constructors`.",
+            "A collision was detected between the following constructors of the class `{$class->type->toString()}`: `$constructors`.",
```
Comment (TimWolla) [inline src/Mapper/Object/Exception/ObjectBuildersCollision.php:23]: Could add braces around `$constructors` for consistency.

File: src/Mapper/Tree/Exception/CannotInferFinalClass.php [hunk 2]
```diff
@@ -14,7 +14,7 @@ final class CannotInferFinalClass extends RuntimeException
     public function __construct(ClassType $class, FunctionDefinition $function)
     {
         parent::__construct(
-            "Cannot infer final class `{$class->className()}` with function `{$function->signature()}`.",
+            "Cannot infer final class `{$class->className()}` with function `$function->signature`.",
```
Comment (TimWolla) [inline src/Mapper/Tree/Exception/CannotInferFinalClass.php:17]: Removal of braces.

File: src/Mapper/Tree/Exception/MissingObjectImplementationRegistration.php [hunk 3]
```diff
@@ -13,7 +13,7 @@ final class MissingObjectImplementationRegistration extends RuntimeException
     public function __construct(string $name, FunctionDefinition $functionDefinition)
     {
         parent::__construct(
-            "No implementation of `$name` found with return type `{$functionDefinition->returnType()->toString()}` of `{$functionDefinition->signature()}`.",
+            "No implementation of `$name` found with return type `{$functionDefinition->returnType->toString()}` of `$functionDefinition->signature`.",
```
Comment (TimWolla) [inline src/Mapper/Tree/Exception/MissingObjectImplementationRegistration.php:16]: Removal of braces.

File: src/Mapper/Tree/Message/MessageBuilder.php [hunk 4]
```diff
@@ -121,18 +121,17 @@ public function parameters(): array
     /**
      * @psalm-pure
      *
-     * PHP8.1 intersection
      * @return MessageType&HasCode&HasParameters
```
Comment (TimWolla) [inline src/Mapper/Tree/Message/MessageBuilder.php:124]: redundant

File: tests/StaticAnalysis/inferring-types-with-plugin.php [hunk 5]
```diff
@@ -53,17 +53,15 @@ function mapping_function_arguments_will_infer_object_of_same_type(ArgumentsMapp
 
 function mapping_static_method_arguments_will_infer_object_of_same_type(ArgumentsMapper $mapper): void
 {
-    // PHP8.1 First-class callable syntax
-    $result = $mapper->mapArguments(Closure::fromCallable([SomeClass::class, 'someStaticMethod']), []);
+    $result = $mapper->mapArguments(Closure::fromCallable(SomeClass::someStaticMethod(...)), []);
```
Comment (TimWolla) [inline tests/StaticAnalysis/inferring-types-with-plugin.php:56]: I believe the `Closure::fromCallable()` is redundant now (same below).

**Review by TimWolla — COMMENTED** [review]
Some remarks. The missing braces around string interpolation also applies to other files that I did not comment explicitly. I also didn't look through everything.

File: src/Mapper/Tree/Message/MessageBuilder.php (hunk 4 above)
Comment (romm) [reply src/Mapper/Tree/Message/MessageBuilder.php:124]: Actually that is not, as `MessageType` is a generic type from the class. Thanks for noticing, though!

File: tests/StaticAnalysis/inferring-types-with-plugin.php (hunk 5 above)
Comment (romm) [reply tests/StaticAnalysis/inferring-types-with-plugin.php:56]: Thanks, will open a small PR for that!

File: src/Mapper/ArgumentsMapperError.php
Comment (romm) [reply src/Mapper/ArgumentsMapperError.php:1]: Thanks for noticing, but I think I'll let it as is, I don't really mind lack of consistency for this kind of things. 😛

**Comment by romm** [comment]
Thanks @TimWolla for the review!

File: src/Mapper/Tree/Message/MessageBuilder.php (hunk 4 above)
Comment (TimWolla) [reply src/Mapper/Tree/Message/MessageBuilder.php:124]: Ah, indeed. I didn't notice the difference between Message and MessageType in the definitions.

**Pull request merged**